### PR TITLE
Execute CI tests not requiring Vault in CircleCI [BT-118]

### PR DESCRIPTION
**Pull request by grsterin**

### CODE REVIEW

**Review by kshakir — APPROVED** [review]
If the CIs are happy I'm happy 🤖

File: src/ci/bin/test.inc.sh [hunk 1]
```diff
@@ -1278,8 +1362,10 @@ cromwell::build::setup_common_environment() {
             cromwell::private::delete_boto_config
             cromwell::private::delete_sbt_boot
             cromwell::private::upgrade_pip
-            cromwell::private::pull_common_docker_images
-            cromwell::private::start_docker_databases
+            if [[ "${BUILD_TYPE}" != "sbt" ]]; then
+              cromwell::private::pull_common_docker_images
+              cromwell::private::start_docker_databases
```
Comment (mcovarr) [inline src/ci/bin/test.inc.sh:1367]: TOL probably some other `BUILD_TYPE`s like metadata comparison and manifest creator also don't care about these

**Review by mcovarr — APPROVED** [review]
LGTM

File: src/ci/bin/test.inc.sh (hunk 1 above)
Comment (grsterin) [reply src/ci/bin/test.inc.sh:1367]: Actually I think there's a chance this if statement is not required at all for CircleCI

Comment (aednichols) [reply src/ci/bin/test.inc.sh:1367]: I feel like the cost of starting the DBs is low and we can reduce complexity by having things be less specific. Just a loosely held thought.

File: src/ci/bin/test.inc.sh [hunk 2]
```diff
@@ -215,6 +218,83 @@ cromwell::private::create_build_variables() {
...
+                circle_minimal_tests=false
+            fi
+
+            echo "Building for commit message='${circle_commit_message}' with force=${circle_force_tests} and minimal=${circle_minimal_tests}"
+
+            # For solely documentation updates run only checkPublish. Otherwise always run sbt, even for 'push'.
+            # This allows quick sanity checks before starting PRs *and* publishing after merges into develop.
+            if [[ "${circle_force_tests}" == "true" ]]; then
+                CROMWELL_BUILD_RUN_TESTS=true
+            elif [[ "${CROMWELL_BUILD_ONLY_DOCS_CHANGED}" == "true" ]] && \
```
Comment (aednichols) [inline src/ci/bin/test.inc.sh:282]: If we meet our goal of making tests fast-ish and reliable we could consider dropping docs-only special handling to reduce complexity

Comment (grsterin) [reply src/ci/bin/test.inc.sh:282]: I agree, but I think it should be a separate follow-up ticket.
We could also remove `force ci` thing.

Comment (aednichols) [reply src/ci/bin/test.inc.sh:282]: Agreed.

**Review by aednichols — APPROVED** [review]
Very nice!

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+      BUILD_TYPE: << parameters.build-type >>
+      BUILD_MYSQL: << parameters.build-mysql >>
+      BUILD_MARIADB: << parameters.build-mariadb >>
+      BUILD_POSTGRESQL: << parameters.build-postgresql >>
+      BUILD_BRANCH_OR_PR: << parameters.branch-or-pr-build >>
+
+workflows:
+  all-tests:
+    jobs:
+      - test:
```
Comment (breilly2) [inline .circleci/config.yml:71]: Would it be too soon to remove these test runs from `.travis.yml`?

**Comment by breilly2** [comment]
In most cases, the `-branch` build has a significantly shorter runtime than the `-pr` build: https://app.circleci.com/pipelines/github/broadinstitute/cromwell/367/workflows/7b1a2a51-80b7-432a-b883-4c28c15741d4. Is the `-branch` build doing the right thing?

**Comment by grsterin** [comment]
> In most cases, the `-branch` build has a significantly shorter runtime than the `-pr` build: https://app.circleci.com/pipelines/github/broadinstitute/cromwell/367/workflows/7b1a2a51-80b7-432a-b883-4c28c15741d4. Is the `-branch` build doing the right thing?

Yes, it mimics current behaviour we have for Travis - for branch builds we actually only run sbt build and other builds just succeed fast without actually running tests, unless there's a `[force ci]` in the commit message

File: .circleci/config.yml (hunk 3 above)
Comment (grsterin) [reply .circleci/config.yml:71]: I think it would be better to remove Travis stuff all at once, after everything runs in CircleCI

**Comment by breilly2** [comment]
Ah, okay. Too bad we can't suppress the build altogether in this case, but I get that we can't do that without also getting rid of the `[force ci]` feature.

**Pull request merged**